tests/Unit/Classes/Db/DbQueryCoreTest.php

```diff
@@ -0,0 +1,186 @@
+ */
+
+declare(strict_types=1);
+
+namespace Tests\Unit\Classes;
+
+if (!defined('_DB_PREFIX_')) {
+    define('_DB_PREFIX_', 'test_');
+}
+
+use DbQuery;
+use PHPUnit\Framework\TestCase;
+
+class DbQueryTest extends TestCase
+{
+    /*
+     * get DbQuery object
+     *
+     * @return DbQuery
+     */
+    private function getDbQueryInstance(): DbQuery
+    {
+        return new DbQuery();
+    }
+
+    /**
+     * @param mixed $type
+     * @param string $expectedType
+     *
+     * @dataProvider providerType
+     */
+    public function testType($type, string $expectedType): void
+    {
+        $dbQuery = $this->dbQueryInstance();
+        $dbQuery->type($type);
+        $this->assertSame($expectedType, $dbQuery->getQuery()['type']);
+    }
+
+    /**
+     * @param mixed $fields
+     * @param array $expectedSelect
+     *
+     * @dataProvider providerSelect
+     */
+    public function testSelect($fields, array $expectedSelect): void
+    {
+        $dbQuery = $this->dbQueryInstance();
+        $dbQuery->select($fields);
+        $this->assertSame($expectedSelect, $dbQuery->getQuery()['select']);
+    }
+
+    /**
+     * @param string $table
+     * @param string|null $alias
+     * @param string $expectedValue
+     *
+     * @dataProvider providerFrom
+     */
+    public function testFrom(string $table, ?string $alias, string $expectedValue): void
+    {
+        $dbQuery = $this->dbQueryInstance();
+        $dbQuery->from($table, $alias);
+        $this->assertSame($expectedValue, $dbQuery->getQuery()['from']);
+    }
+
+    /**
+     * @param mixed $dbQuery
+     * @param mixed $expectedValue
+     *
+     * @dataProvider providerBuild
+     */
+    public function testBuild(DbQuery $dbQuery, $expectedValue)
+    {
+        $this->assertSame(trim($dbQuery->build()), trim($expectedValue));
+    }
+
+    public function providerType(): array
+    {
+        return [
+            ['SELECT', 'SELECT'],
+            ['DELETE', 'DELETE'],
+            ['INVALID_TYPE', 'SELECT'],
+            ['select', 'SELECT'],
+            ['delete', 'SELECT'],
+            [666, 'SELECT'],
+            [false, 'SELECT'],
+            [null, 'SELECT'],
+        ];
+    }
+
+    public function providerSelect(): array
+    {
+        return [
+            ['FIELD1', [
+                0 => 'FIELD1',
+            ]],
+            ['FIELD1, FIELD2', [
+                0 => 'FIELD1, FIELD2',
+            ]],
+            [null, []],
+            [false, []],
+        ];
+    }
+
+    public function providerFrom(): array
+    {
+        return [
+            ['table_name', 'alias', [
+                0 => '`' . _DB_PREFIX_ . 'table_name` alias',
+            ]],
+            ['table_name', null, [
+                0 => '`' . _DB_PREFIX_ . 'table_name`',
+            ]],
+        ];
+    }
+
+    public function providerBuild(): array
+    {
+        define('_BREAK_LINE_', "\n");
```

Comment:
Use class const instead, defining a constant is useless :)

Reply:
This is also something you can refactor in the DbQueryCore class by using PHP_EOL or another const